From 960d665bf54dd4cf403fc1d6684f0cc5fa562e46 Mon Sep 17 00:00:00 2001
From: Christian Bruel <christian.bruel@foss.st.com>
Date: Tue, 6 Jun 2023 11:23:34 +0200
Subject: [PATCH] arm64: dts: st: Add PCIe EP on the stm32mp257f-ev1 board

Configure PCIe EP and the ComboPHY with external pad clock on
MP25 eval board to be used as a device. Exclusive with pcie_rc
that is the default.
The EP is configured for CommonRef clock, need to deassert CLKREF#.
Note that the HW board needs to be modified to root the PCIE_CLK pins
to PCIE_CLKIN.

Signed-off-by: Christian Bruel <christian.bruel@foss.st.com>
Change-Id: I8acf6faa28921be9a70b5b3a0550a149e0b2b6ca
---
 arch/arm64/boot/dts/st/stm32mp257f-ev1.dts | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts b/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
index 63625f153eaf..6bc9c22ced6e 100644
--- a/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
@@ -502,6 +502,14 @@ flash0: flash@0 {
 	};
 };
 
+&pcie_ep {
+	pinctrl-names = "default", "init";
+	pinctrl-0 = <&pcie_pins_a>;
+	pinctrl-1 = <&pcie_init_pins_a>;
+	reset-gpios = <&gpioj 8 GPIO_ACTIVE_LOW>;
+	status = "disabled";
+};
+
 &pcie_rc {
 	pinctrl-names = "default", "init";
 	pinctrl-0 = <&pcie_pins_a>;
-- 
2.39.5

